Add Typing fixes and other code changes

ch3ck opened the pull request:
# Proposed changes

- Enables mypy checks
- Code fixes for mypy

danielfernau commented on protect_archiver/client/__init__.py:
```diff
@@ -13,7 +17,7 @@ def __init__(
         protocol: str = Config.PROTOCOL,
         username: str = Config.USERNAME,
-        password: str = Config.PASSWORD,
+        password: Optional[str] = Config.PASSWORD,
```
I'm not sure if the password variable can ever be considered optional when interacting with a UniFi Protect Server...?

ch3ck replied:
Yes, that's because in `Config.PASSWORD` it's default to none, so it won't be correct setting the password type to `str` since we'll have to force a default of `""` which can be dangerous, so `Optional[str]` is the best type to represent the password here

danielfernau commented on protect_archiver/client/legacy.py:
```diff
@@ -14,16 +16,16 @@ def __init__(
 
-        self._access_key = None
-        self._api_token = None
+        self._access_key: Optional[str] = None
+        self._api_token: Optional[str] = None
```
Same as in the other location; what's your idea behind marking these as optional? Maybe I'm missing something...

ch3ck replied:
@danielfernau same explanation as above, we clearly see here it defaults to None if even though we might expect a string type hence `Optional[str]`

danielfernau replied:
@ch3ck That makes sense, thanks for the explanation!

danielfernau commented on protect_archiver/client/unifi_os.py:
```diff
@@ -23,7 +25,7 @@ def __init__(
         self.verify_ssl = verify_ssl
 
         self._access_key = None
-        self._api_token = None
+        self._api_token: Optional[str] = None
```
One optional, the other one not?

ch3ck replied:
Same as above

danielfernau commented on protect_archiver/config.py:
```diff
@@ -1,12 +1,15 @@
     PROTOCOL: str = "https"
     USERNAME: str = "ubnt"
-    PASSWORD: str = None
+    PASSWORD: Optional[str] = None
```
Same here. I don't think any of the authentication variables can ever be considered optional.

ch3ck replied:
same as above

danielfernau reviewed (commented):
Looks very good to me so far, just added a few comments 👍 
If you could enlighten me what your thoughts are behind marking some of the authentication related variables Optional?

(Also, there's a protect_archiver/.DS_Store file that somehow ended up in the PR 😄)

ch3ck commented:
> Looks very good to me so far, just added a few comments 👍 If you could enlighten me what your thoughts are behind marking some of the authentication related variables Optional?
> 
> (Also, there's a protect_archiver/.DS_Store file that somehow ended up in the PR 😄)

Nice catch, fixed!

danielfernau commented:
Perfect. Thanks for your help. Merged!
I'll see that I get around to releasing a new version including all these changes later this week.

danielfernau commented:
Hey @ch3ck -- [v2.1.0](https://github.com/danielfernau/unifi-protect-video-downloader/releases/tag/v2.1.0) just went live. Didn't want to wait too much longer, and everything else has been moved to milestone [v2.2.0](https://github.com/danielfernau/unifi-protect-video-downloader/milestone/5) for now. Thanks again for your help and contributions to this version!
